src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_resolver.py

```diff
@@ -260,6 +310,37 @@ def resolve(self, var, attribute):
         except:
             return getattr(var, attribute)
 
+    def get_contents_debug_adapter_protocol(self, lst):
+        '''
+        This method is to be used in the case where the variables are all saved by its id (and as
+        such don't need to have the `resolve` method called later on, so, keys don't need to
+        embed the reference in the key).
+
+        Note that the return should be ordered.
+
+        :return list(tuple(name:str, value:object, evaluateName:str))
+        '''
+        l = len(lst)
+        ret = []
+
+        format_str = '%0' + str(int(len(str(l)))) + 'd'
```

Comment:
This looks like it needs to be `l-1` (i.e. index of the last element, rather than the length).

Reply:
Also, this is always using `%d`, but repr for keys is expected to respect hex mode as well.

Reply:
Fixed the `l-1` (with proper tests).

Created https://github.com/Microsoft/ptvsd/issues/1174 to deal with the hex formatting.